File: 898-transpose-matrix/transpose-matrix.cpp
class Solution {
public:
    vector<vector<int>> transpose(vector<vector<int>>& matrix) {
        int n = matrix.size();        // Rows
        int m = matrix[0].size();     // Columns

        vector<vector<int>> result(m, vector<int>(n));  // m rows, n cols

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                result[j][i] = matrix[i][j];
            }
        }

        return result;
    }
};

File: 898-transpose-matrix/transpose-matrix.java
class Solution {
    public int[][] transpose(int[][] matrix) {
        int m = matrix.length;       // rows
        int n = matrix[0].length;    // cols

        int[][] result = new int[n][m];  // note: flipped dimensions

        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                result[j][i] = matrix[i][j];
            }
        }

        return result;
    }
}
